build.gradle

```diff
@@ -70,24 +70,98 @@ allprojects {
+                'BadComparable',
+                'BadInstanceof',
+                'BoxedPrimitiveConstructor',
+                'CatchFail',
+                'ClassCanBeStatic',
+                'CompareToZero',
+                'ComplexBooleanConstant',
+                'ConstantField',
+                'DefaultCharset',
+                'DeprecatedThreadMethods',
+                'DoubleBraceInitialization',
+                'DoubleCheckedLocking',
+                'EqualsIncompatibleType',
+                'EqualsUnsafeCast',
+                'FallThrough',
+                'FieldCanBeFinal',
+                'FieldMissingNullable',
+                'FunctionalInterfaceClash',
+                'FutureReturnValueIgnored',
+                'InstanceOfAndCastMatchWrongType',
+                'IntLongMath',
+                'JavaInstantGetSecondsGetNano',
+                'JavaTimeDefaultTimeZone',
+                'JdkObsolete',
+                'LockNotBeforeTry',
+                'LogicalAssignment',
+                'MethodCanBeStatic',
+                'MissingOverride',
+                'MixedMutabilityReturnType',
+                'ModifyCollectionInEnhancedForLoop',
+                'NonCanonicalStaticMemberImport',
+                'NonOverridingEquals',
+                'NullAway',
+                'NullableVoid',
+                'ObjectsHashCodePrimitive',
+                'ParameterName',
+                'ParameterNotNullable',
+                'PreconditionsCheckNotNullRepeated',
+                'PreferCollectionTransform',
+                'PreferListsPartition',
+                'PreferSafeLoggableExceptions',
+                'PreferSafeLoggingPreconditions',
+                'PrivateConstructorForUtilityClass',
+                'ReturnMissingNullable',
+                'RxReturnValueIgnored',
+                'ShortCircuitBoolean',
+                'Slf4jLogsafeArgs',
+                'StaticGuardedByInstance',
+                'StreamResourceLeak',
+                'SwitchStatementDefaultCase',
+                'SynchronizeOnNonFinalField',
+                'ThreadJoinLoop',
+                'ToStringReturnsNull',
+                'UnnecessaryParentheses',
+                'UnusedNestedClass',
+                'UnusedVariable',
+            ]
+
+            for (String checkName : warningsToPromote) {
+                options.errorprone.errorproneArgs << '-Xep:' + checkName + ':ERROR'
+            }
+
+            logger.info("Error-prone args: {}", options.errorprone.errorproneArgs)
+
+            if (generateErrorPronePatch) {
+                def patchChecks = [
+                    "PreferSafeLoggableExceptions",
+                    "PreferCollectionTransform",
+                    "PreferListsPartition",
+                ]
+
+                logger.info("Generating error-prone.patch for checks {} apply with:\n"
+                        + "    find . -maxdepth 3 -type f -name 'error-prone.patch' \\ \n"
```

Comment:
I think `-maxdepth 2` is sufficient here, though it doesn't make much difference. Some day I'll get around to writing a proper gradle task for these.

Reply:
Yeah, maxdepth 2 works for tritium.

Adding a configurable set of checks and option to generate patch to gradle-Baseline would be useful. Might give that a shot if I can find some time.